pkg/references/object_key.go

```diff
@@ -0,0 +1,34 @@
+package references
```

Comment:
All three of these functions all do the exact same thing, fetch `vxRef.Namespace` `vxRef.Name`  and return it. Can't all `VirtualReferences` have an interface that allows to to `getNamespace()` and  `getName()`  then this would just be `func ObjectKeyForVirtualReference(obj metav1.Object, vRef appmesh.VirtualReference) types.NamespaceName`

Reply:
yeah, we can.
But I don't want to pollution the type definitions of these references for now.  (i have been planning to extend these reference to reference external objects outside of k8s)
I think we can aggregate these references once we finalized everything :D

Reply:
We should add a follow up issue to track this.

Reply:
sure :D these tasks are post-GA tasks :D